### PR TITLE
GlobalExceptionHandler and isbn validation

### PR DESCRIPTION
**Pull request by Stelmaschyk**

### CODE REVIEW

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -37,14 +38,14 @@ public BookDto getBookById(@PathVariable Long id) {
 
     @PostMapping
     @ResponseStatus(HttpStatus.CREATED)
-    public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
+    public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/controller/BookController.java:41]: You can also validate `@PathVariable @Positive Long id` 
add `@Validated` under the controller in such case

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    ) {
+
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java]:
```suggestion
        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 3]
```diff
@@ -23,13 +26,20 @@ public class Book {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     @Column(nullable = false)
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]: `@Column(nullable = false)` is enough for validation, remove `@NotNull` everywhere

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 4]
```diff
@@ -23,13 +26,20 @@ public class Book {
...
     private String author;
     @Column(nullable = false, unique = true)
+    @NotNull
+    @Isbn
     private String isbn;
     @Column(nullable = false)
+    @NotNull
+    @Min(0)
     private BigDecimal price;
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 5]
```diff
@@ -23,13 +26,20 @@ public class Book {
...
     private String title;
     @Column(nullable = false)
+    @NotNull
     private String author;
     @Column(nullable = false, unique = true)
+    @NotNull
+    @Isbn
     private String isbn;
     @Column(nullable = false)
+    @NotNull
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]:
```suggestion
    @Column(nullable = false)
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+            WebRequest request
+    ) {
+
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java]: You can save strings to constants

File: src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        return new ResponseEntity<>(body, headers, HttpStatus.BAD_REQUEST);
+    }
+
+    private String getErrorMessage(ObjectError e) {
+        if (e instanceof FieldError) {
+            String field = ((FieldError) e).getField();
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/exception/CustomGlobalException.java]:
```suggestion
        if (e instanceof FieldError fieldError) {
            String field = fieldError.getField();
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 8]
```diff
@@ -27,8 +29,10 @@ public class Book {
     @Column(nullable = false)
     private String author;
     @Column(nullable = false, unique = true)
+    @Isbn
     private String isbn;
     @Column(nullable = false)
+    @Min(0)
```
Comment (MentorMateAcademy) [inline src/main/java/mate/academy/bookstore/model/Book.java]: Use Positive anno

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 9]
```diff
@@ -27,8 +29,10 @@ public class Book {
     @Column(nullable = false)
     private String author;
     @Column(nullable = false, unique = true)
+    @Isbn
     private String isbn;
     @Column(nullable = false)
+    @Positive
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]:
```suggestion
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 10]
```diff
@@ -27,8 +29,10 @@ public class Book {
     @Column(nullable = false)
     private String author;
     @Column(nullable = false, unique = true)
+    @Isbn
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]: it should be validated on DTO layer
```suggestion
```

File: src/main/java/mate/academy/bookstore/model/Book.java [hunk 11]
```diff
@@ -27,8 +29,10 @@ public class Book {
     @Column(nullable = false)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/model/Book.java]: Where is validation on CreateBookRequestDto ?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

File: src/main/java/mate/academy/bookstore/controller/BookController.java [hunk 12]
```diff
@@ -21,6 +23,7 @@
 @RequiredArgsConstructor
 @RestController
 @RequestMapping("/books")
+@Validated
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/controller/BookController.java]: This annotation is needed if you use `@Positive` for `@PathVariable Long id` for example, if not - you can remove it

File: src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java [hunk 13]
```diff
@@ -1,14 +1,24 @@
...
 @Data
 public class CreateBookRequestDto {
+    @NotBlank
+    @Size(max = 40)
     private String title;
+    @NotBlank
     private String author;
+    @Isbn
     private String isbn;
+    @Positive
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/dto/CreateBookRequestDto.java:21]:
```suggestion
    @Positive
    @NotNull
```

File: src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java [hunk 14]
```diff
@@ -1,15 +1,25 @@
...
 public class UpdateBookRequestDto {
     private Long id;
+    @NotBlank
+    @Size(max = 40)
     private String title;
+    @NotBlank
     private String author;
+    @Isbn
     private String isbn;
+    @Positive
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java:21]:
```suggestion
    @Positive
    @NotNull
```

File: src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java [hunk 15]
```diff
@@ -1,15 +1,25 @@
...
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Positive;
+import jakarta.validation.constraints.Size;
 import java.math.BigDecimal;
 import lombok.Data;
+import mate.academy.bookstore.validation.Isbn;
 
 @Data
 public class UpdateBookRequestDto {
     private Long id;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/bookstore/dto/UpdateBookRequestDto.java]:
```suggestion
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see comments

**Review by Elena-Bruyako — APPROVED** [review]
well done

**Pull request merged**